packages/eslint-plugin/docs/rules/member-ordering.md

```diff
@@ -2,52 +2,63 @@
-
-### Sorting members
-
-Besides grouping the members and sorting their groups, this rule also allows to sort the members themselves (e.g. `a`, `b`, `c`, ...). You have 2 options: Sort all of them while ignoring their type or sort them while respecting their types (e.g. sort all fields in an interface alphabetically).
-
 ## Options
 
-These options allow to specify how to group the members and sort their groups.
-
-- Sort groups, don't enforce member order: Use `memberTypes`
-- Sort members, don't enforce group order: Use `order`
-- Sort members within groups: Use `memberTypes` and `order`
-
 ```ts
-type SortedOrderConfig = {
-  memberTypes?: MemberType[] | 'never';
-  order: 'alphabetically' | 'alphabetically-case-insensitive' | 'as-written';
-};
-
-type OrderConfig = MemberType[] | SortedOrderConfig | 'never';
-
-type Options = {
+interface Options {
   default?: OrderConfig;
   classes?: OrderConfig;
   classExpressions?: OrderConfig;
   interfaces?: OrderConfig;
   typeLiterals?: OrderConfig;
-};
+}
+
+type OrderConfig = MemberType[] | SortedOrderConfig | 'never';
+
+interface SortedOrderConfig {
+  memberTypes?: MemberType[] | 'never';
+  order: 'alphabetically' | 'alphabetically-case-insensitive' | 'as-written';
+}
+
+// See below for the more specific MemberType strings
+type MemberType = string | string[];
 ```
 
-See below for the possible definitions of `MemberType`.
+You can configure `OrderConfig` options for:
+
+- **`default`**: all constructs (used as a fallback)
+- **`classes`**?: override ordering specifically for classes
+- **`classExpressions`**?: override ordering specifically for class expressions
+- **`interfaces`**?: override ordering specifically for interfaces
+- **`typeLiterals`**?: override ordering specifically for type literals
+
+The `OrderConfig` settings for each kind of construct may configure sorting on one or both two levels:
+
+- **`memberType`**: organizing on member type groups such as methods vs. properties
+- **`order`**: organizing based on member names, such as alphabetically
+
+### Groups
+
+You can define many different groups based on different attributes of members.
+The supported member attributes are, in order:
 
-### Deprecated syntax
+- **Accessibility** (`'public' | 'protected' | 'private'`)
+- **Decoration** (`'decorated'`): Whether the member has an explicit accessibility decorator
+- **Kind** (`'call-signature' | 'constructor' | 'field' | 'get' | 'method' | 'set' | 'signature'`)
 
-Note: There is a deprecated syntax to specify the member types as an array.
+Member attributes may be joined with a `'-'` to combine into more specific groups.
+For example, `'public-field'` would come before `'private-field'`.
 
-### Member types (granular form)
+<details>
+  <summary>Expand this to see the full list of all supported member type groups.</summary>
```

Comment:
This exposes a bug(?) in Docusaurus. The right-side headings menu still displays headings within a `<details>` ([which should be allowed](https://a11ysupport.io/tech/html/summary_element)) even if collapsed. Clicking those heading links does nothing while collapsed.

https://user-images.githubusercontent.com/3335181/165952094-104920b3-5f1d-4ed6-9302-853712975b05.mp4

Reply:
If the bug isn't accepted -or if you the reviewer hate this pattern- I can remove the `<details>` and move its content to a new section after `## Examples`. Otherwise I'd consider that bug blocking merging this PR. 

The rest of the PR should still be ready for review though! ✨